pulser-core/pulser/sequence/sequence.py

```diff
@@ -803,6 +824,7 @@ def declare_channel(
             if not self._in_xy:
                 self.set_magnetic_field()
                 self._in_xy = True
+                self._in_ising = False
```

Comment:
Why do we need this line? If it's just to make sure, I would put it before `self._in_xy=True`

Reply:
Unuseful indeed, I just thought it would make things clearer (like an assert). Let me delete it.